pandas/io/excel/_calamine.py

```diff
@@ -107,10 +106,16 @@ def _convert_cell(value: _CellValue) -> Scalar | NaTType | time:
                     return val
                 else:
                     return value
+            elif isinstance(value, datetime):
+                # Return datetime.datetime as-is to match openpyxl behavior (GH#59186)
+                return value
             elif isinstance(value, date):
-                return pd.Timestamp(value)
+                # Convert date to datetime to match openpyxl behavior (GH#59186)
+                return datetime(value.year, value.month, value.day)
             elif isinstance(value, timedelta):
-                return pd.Timedelta(value)
+                # Return datetime.timedelta as-is to match openpyxl behavior (GH#59186)
+                # (previously returned pd.Timedelta)
+                return value
```

Comment:
Can you combine this branch with the `isinstance(value, datetime)` branch by combining the isinstance checks?